st2common/st2common/bootstrap/triggersregistrar.py

```diff
@@ -0,0 +1,182 @@
+# Licensed to the StackStorm, Inc ('StackStorm') under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+import six
+
+from st2common import log as logging
+from st2common.constants.meta import ALLOWED_EXTS
+from st2common.bootstrap.base import ResourceRegistrar
+import st2common.content.utils as content_utils
+from st2common.models.api.trigger import TriggerTypeAPI
+from st2common.persistence.trigger import TriggerType
+
+__all__ = [
+    'TriggersRegistrar',
+    'register_triggers'
+]
+
+LOG = logging.getLogger(__name__)
+
+
+class TriggersRegistrar(ResourceRegistrar):
```

Comment:
Base resource registrar class definitely needs some more love in the future :)

It's better than it was originally, but there is still way too much duplicated code.

Anyway, nothing for you to do right now, more of a note for us to fix / improve it.